server.js

```diff
@@ -164,24 +166,61 @@ function resolveWalkthroughLocations(locations) {
         return reject(new Error(`Invalid location ${location}`));
       } else if (isPath(location)) {
         console.log(`Importing walkthrough from path ${location}`);
-        const locationResult = Object.assign({}, locationResultTemplate, { local: location });
+        const locationResult = Object.assign({ parentId: 'local' }, locationResultTemplate, { local: location });
         return resolve(locationResult);
       } else if (isGitRepo(location)) {
-        console.log(`Importing walkthrough from git ${location}`);
         const clonePath = path.join(TMP_DIR, tmpDirPrefix);
+
+        // Need to parse out query params for walkthroughs, e.g custom directory
+        const cloneUrl = generateCloneUrlFromLocation(location)
+        const repoName = retWalkthroughRepoNameFromLocation(location)
```

Comment:
Including this will prevent clashes in IDs across walkthroughs, even if they use the same ID, e.g `walkthrough-1` or `lab01`.

Reply:
is that just for one repo, or all repos specified in walkthrough_locations?

Reply:
@evanshortiss ping re. this question

Reply:
@finp do you mean can two walkthroughs in two repos use the same ID? If so, then yes. For example

```
-- REPO_1
  -- walkthroughs
    -- lab01

-- REPO_2
  -- walkthroughs
    -- lab01
```

Would work since the generated ID is `REPO_1-lab01` and `REPO_2-lab01`

Reply:
Oh. I see what you mean. Yes, if it's a single repo with multiple folder then it also works, e.g

```
-- REPO
  |-- set-a
     |-- lab01
  |-- set-b
     |-- lab01
```

Would generate an ID `REPO-set-a-lab01` or similar.

Reply:
thanks for the clarification, and yes, this does address the original Jira ;)